### PR TITLE
[bugfix] Handle numeric timezones in RFC2822 parsing, fixes #4082, #3920

### LINKED ISSUE #4082: Wrong "Time from now" for RFC 2822 date
Moment.js version 1.18, proper RFC 2822 date time format show wrong value for `moment().fromNow()`

**Steps to Reproduce:**

```js
var DATE_RFC2822 = "ddd, DD MMM YYYY HH:mm:ss ZZ";
var now = moment().format(DATE_RFC2822);
console.log(now); // Mon, 24 Jul 2017 10:00:47 +0300
console.log(moment(now).fromNow()); // in 3 hours
console.log(moment(new Date(now)).fromNow()); // a few seconds ago
```
both must be `a few seconds ago`

### PR DESCRIPTION
**Pull request by ashsearle**

### CODE REVIEW

**Comment by maggiepint** [comment]
Thanks for your contribution. Can we get unit tests for this? Also, I think that we may need to consider validation for the input string.

**Comment by ashsearle** [comment]
@maggiepint I hope the new unit-tests cover the cases you were interested in.

**Comment by ichernev** [comment]
Is this related to https://github.com/moment/moment/pull/3938

File: src/lib/create/from-string.js [hunk 1]
```diff
@@ -139,17 +139,16 @@ export function configFromRFC2822(config) {
...
             case 2: // military
-                if (timezoneIndex === 0) {
-                    timezone = ' +0000';
-                } else {
-                    timezoneIndex = military.indexOf(match[5][1].toUpperCase()) - 12;
-                    timezone = ((timezoneIndex < 0) ? ' -' : ' +') +
-                        (('' + timezoneIndex).replace(/^-?/, '0')).match(/..$/)[0] + '00';
-                }
+                timezoneIndex = military.indexOf(match[5][1].toUpperCase()) - 12;
+                timezone = ((timezoneIndex < 0) ? ' -' : ' +') +
```
Comment (ichernev) [inline src/lib/create/from-string.js]: How about we use utils/zero-fill (timezoneIndex, 2, true) instead of this hack

**Comment by ichernev** [comment]
For the unit tests -- you can add a few military cases, positive and negative and zero. Also make sure to add partial-hour offsets in the mix.

**Comment by ashsearle** [comment]
@ichernev I hadn't seen #3938, but yes: it looks like I'm fixing the same piece of code.

I've added a few more tests to cover the +/- military cases, half-hour and quarter-hour timezones.

(I've avoided redundantly testing zero military timezone 'z' test as it's already used in other unit-tests.)

**Comment by ichernev** [comment]
@ashsearle can you close this issue if everything is working now OR rewrite it on top of latest code.

**Comment by ashsearle** [comment]
Closing as obsolete following #4108

**Pull request closed without merging**